src/book/03-guides/02-tools/03-webots.mdx

```diff
@@ -0,0 +1,92 @@
+---
+section: Guides
+chapter: Tools
+title: Setting Up Webots
+description: Setup instructions for Webots.
+slug: /guides/tools/webots-setup
+---
+
+Webots is a physics environment created by Cyberbotics. This guide details how to set up Webots. It gives instructions on setting up Ubuntu (WSL) inside Windows for Windows users, and the rest of the guide is aimed at Ubuntu users (including Windows WSL Ubuntu users).
+
+## Prerequisites
+
+<details>
+
+<summary>Windows</summary>
+
+1. Follow the instructions for setting up Ubuntu in WSL on the [Getting Started NUbook page under the Windows tab in the Docker section](/guides/main/getting-started#docker).
+
+2. Since Webots uses a GUI, we will need to set up a way to launch and display the program. Use [these instructions](https://medium.com/@chuckdries/installing-gitkraken-in-wsl-2-15bf6459f823) which is for a similar scenario, but with GitKraken instead of Webots. You can ignore installing and launching GitKraken. Instead, we will use the Webots instructions later in this guide to install and launch Webots.
+
+   <Alert type='info'>
+
+   To be able to copy and paste into the Ubuntu terminal, right-click on the bar at the top of the Ubuntu window. Then click on **Properties**, and enable **Use Ctrl+Shift+C/V as Copy/Paste** in the **Options** tab. Note that this is <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>C</kbd>/<kbd>V</kbd> not <kbd>Ctrl</kbd>+<kbd>C</kbd>/<kbd>V</kbd>.
+
+   </Alert>
+
+3. In Ubuntu run
+
+   ```sh
+   sudo apt install libxcb-icccm4 libxcb-image0 libxcb-keysyms1 libxcb-render-util0 libxcb-xfixes0 libxcb-xinerama0 libxcb-xkb1 libxkbcommon-x11-0 libxkbcommon0
+   ```
+
+Follow the rest of the steps on this page within Ubuntu.
+
+</details>
+
+## Install Webots
+
+To install Webots, use the instructions on the [Webots GitHub repository](https://github.com/cyberbotics/webots/wiki/Linux-installation). Follow these instructions inside Ubuntu (WSL) if you are a Windows user. See the previous section on how to set this up. The optional dependencies are not required.
+
+## Get and Build the NUbots Environment
+
+1. Open a terminal. Clone the [NUbots/Webots](https://github.com/NUbots/Webots/) repository and move into it by running
+
+   ```sh
+   git clone https://github.com/NUbots/Webots
+   cd Webots
+   ```
+
+2. Compile the controllers with cmake
+
+   1. Make a build folder and move into it
+
+   ```sh
+   mkdir build && cd build
+   ```
+
+   2. Configure and generate the build files
+
+   ```sh
+   cmake ..
+   ```
+
+   3. Compile the code
+
+   ```sh
+   make
+   ```
+
+## Running Webots
+
+1. Launch Webots by running
```

Comment:
This needs to be performed from the Webots source folder. This may not be clear given the previous section is talking about the NUbots controller repo